scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobDownloader.java

```diff
@@ -0,0 +1,222 @@
+import com.microsoft.rest.v2.util.FlowableUtil;
+
+import io.reactivex.Single;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 13/09/2018
+ */
+public class AzureBlobDownloader extends JavaExecutable {
+
+    private String outputPath;
+
+    private String containerName;
+
+    private String blobName;
+
+    private String accountName;
+
+    private String accountKey;
+
+    private static final String OUTPUT_PATH = "outputPath";
+
+    private static final String CONTAINER_NAME = "containerName";
+
+    private static final String BLOB_NAME = "blobName";
+
+    private ContainerURL containerURL;
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+
+        if (args.containsKey(CONTAINER_NAME) && !args.get(CONTAINER_NAME).toString().isEmpty()) {
+            containerName = args.get(CONTAINER_NAME).toString();
+        } else {
+            throw new IllegalArgumentException("You have to specify a container name. Empty value is not allowed.");
+        }
+
+        if (args.containsKey(BLOB_NAME) && !args.get(BLOB_NAME).toString().isEmpty()) {
+            blobName = args.get(BLOB_NAME).toString();
+        }
+
+        if (args.containsKey(OUTPUT_PATH) && !args.get(OUTPUT_PATH).toString().isEmpty()) {
+            outputPath = args.get(OUTPUT_PATH).toString();
+        } else {
+            //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+            outputPath = getLocalSpace();
+        }
+        // Retrieve the credentials
+        accountName = getThirdPartyCredential("ACCOUNT_NAME");
+        accountKey = getThirdPartyCredential("ACCOUNT_KEY");
+        if (accountName == null || accountKey == null) {
+            throw new IllegalArgumentException("You first need to add your account name and account key (ACCOUNT_NAME, ACCOUNT_KEY) to the third party credentials.");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+
+        File file = new File(outputPath);
+        containerURL = AzureStorageConnectorUtils.createContainerURL(accountName, accountKey, containerName);
+        //download a single blob
+        if (blobName != null && !blobName.isEmpty()) {
+            //check weather or not the outputPath is a folder path or a file path
+            if ((outputPath.lastIndexOf('/') == outputPath.length() - 1)) {
+                createDirIfNotExists(file);
+                String azureBlobLocalRelativePath = Paths.get(outputPath, Paths.get(blobName).getFileName().toString())
+                                                         .toString();
+                downloadBlob(new File(azureBlobLocalRelativePath), blobName);
+            } else {
+                downloadBlob(file, blobName);
```

Comment:
If you use the Optional<String> type you can do : 
```java
downloadBlob(new File(azureBlobLocalRelativePath), blobName.get());
            } else {
                downloadBlob(file, blobName;get());
```
Otherwise it is not necessary to provide an attribute as a parameter to a private method. Indeed the private method have access to this attribute.